### PR TITLE
Slot/Fill pattern with Toolbar

### PR DESCRIPTION
**Pull request by marecar3**
### Description

This PR resolves part of the [Toolbar issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/55). It adds support for Slot/Fill pattern with Toolbar which results in porting list item Toolbar on the bottom of the screen (or on top of the keyboard).

## Slot/Fill pattern with Toolbar 
What's done : 
- [x] Fix : is block focused logic on Android platform
- [x] Implement Slot/Fill "portal" pattern in gutenberg mobile
- [x] Test Slot/Fill pattern with existing InlineToolbar
- [x] Create Toolbar container with Format Toolbar Slot and Heading Toolbar Slot on top of the screen
- [x] Test Slot/Fill pattern with Format Toolbar Slot and existing Format Toolbar

## Context (Consumer, Provider) pattern with Toolbar

- [x] Provide isSelected logic to Toolbar level

screen shot 2018-10-26 at 2 19 54 pm

### How has this been tested?

It is tested against this [PR](https://github.com/WordPress/gutenberg/pull/11115/)

### CODE REVIEW

File: src/block-management/block-manager.js [hunk 1]
```diff
@@ -51,9 +52,24 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	onBlockHolderPressed( clientId: string ) {
+		this.focusDataSoruceItem(clientId);
 		this.props.focusBlockAction( clientId );
 	}
 
+	focusDataSoruceItem(clientId: string) {
```
Comment (koke) [inline src/block-management/block-manager.js]:
```suggestion
	focusDataSourceItem(clientId: string) {
```

File: src/block-management/block-manager.js [hunk 2]
```diff
@@ -51,9 +52,24 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	onBlockHolderPressed( clientId: string ) {
+		this.focusDataSoruceItem(clientId);
```
Comment (koke) [inline src/block-management/block-manager.js]:
```suggestion
		this.focusDataSourceItem(clientId);
```

**Comment by marecar3** [comment]
Update: I am fixing the tests.

**Comment by koke** [comment]
> Use this branch as gutenberg submodule

If you have your `gutenberg` directory with that branch (or any other commit checked out), you should see changes in `gutenberg` when doing `git status`. You can add and commit those changes so anyone checking out this PR will already have the right Gutenberg version

File: src/block-management/block-manager.js (hunk 1 above)
Comment (marecar3) [reply src/block-management/block-manager.js]: Fixed.

**Comment by hypest** [comment]
Nice work @marecar3 !

Tried the PR locally and it seems something goes wrong with the number by the heading buttons (check out the "2", the "3" and the "4" in the screenshot below):

screenshot-1540803959602

**Comment by gziolo** [comment]
You will have to wrap every block with `BlockEdit` component (https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/block-edit/index.js) and feed it with the proper value of `isSelected` based on store.

File: src/block-management/block-holder.js [hunk 3]
```diff
@@ -64,13 +65,15 @@ export default class BlockHolder extends React.Component<PropsType, StateType> {
 		}
 
 		return (
-			<Block
+			<BlockEdit
```
[the remark this answers is not in the document]
Comment (marecar3) [reply src/block-management/block-holder.js]: Thanks @gziolo !

**Comment by diegoreymendez** [comment]
@marecar3 - I'm removing my assignment for the time being (for transparency) as I won't be able to focus on this PR for at least a few days.  Sorry!

**Comment by marecar3** [comment]
Hey @diegoreymendez , thanks for the update!

**Pull request merged**